frontend/src/components/ContentPanel/ContentPanelPrompts/PromptIOBox/IOBoxBar/IOBoxButton/index.tsx

```diff
@@ -25,18 +21,31 @@ export const IOBoxButton: React.FC<IOBoxButtonProps> = ({
     visible,
     errors,
 }) => {
+    const [nameVisible, setNameVisible] = useState(false);
+
     return (
-        <ExtendedFAB
-            name={name}
-            icon={icon ?? 'PlusIcon'}
-            colorPalette={colorPalette}
-            disabled={errors ? true : false}
-            onClick={onClick}
+        <div
+            data-testid={`iobox-${name}`}
+            onMouseEnter={() => setNameVisible(true)}
+            onMouseLeave={() => setNameVisible(false)}
             className={classNames(
-                'transition-all',
-                { '-translate-y-16': visible },
-                { '': !visible }
+                'transition-all mb-0 mr-0',
+                { '-translate-y-[11.5em]': visible },
+                { '': !visible },
+                { 'mr-0 scale-105': nameVisible }
             )}
-        />
+        >
+            <FilledButton
+                name={nameVisible ? name : ''}
+                icon={icon}
+                colorPalette={colorPalette}
+                disabled={errors ? true : false}
```

Comment:
Would
errors !== undefined //returns true for an empty string
or
Boolean(errors) //exactly same behaviour in this contexts
be prettier?
be

Reply:
Removed errors as it is not used by IOBoxButton. The button is not visible if there are errors, so there is no reason to disable it.